source/particle/property/melt_particle.cc

```diff
@@ -0,0 +1,102 @@
+/*
+  Copyright (C) 2015 - 2016 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/particle/property/melt_particle.h>
+#include <aspect/simulator.h>
+#include <aspect/melt.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      template <int dim>
+      void
+      Melt<dim>::initialize_one_particle_property(const Point<dim> &position,
+                                                  std::vector<double> &data) const
+      {
+        data.push_back(0.0);
+      }
+
+      template <int dim>
+      void
+      Melt<dim>::update_one_particle_property(const unsigned int data_position,
+                                              const Point<dim> &,
+                                              const Vector<double> &solution,
+                                              const std::vector<Tensor<1,dim> > &,
+                                              const ArrayView<double> &data) const
+      {
+        AssertThrow(this->introspection().compositional_name_exists("porosity"),
+                    ExcMessage("Material model Melt simple with melt transport only "
+                               "works if there is a compositional field called porosity."));

```

Comment:
Please update the error message inside of the call to `ExcMessage` to the current plugin.